torch/csrc/lazy/ts_backend/ir_builder.h

```diff
@@ -36,28 +36,30 @@ namespace lazy {
 
 struct TorchScriptIrBuilder: IrBuilder {
   NodePtr MakeDeviceData(const std::shared_ptr<BackendData>& data) const override { return DeviceData::Create(data); }
-  NodePtr MakeScalar(const at::Scalar& value, const at::ScalarType& type) const override { return MakeNode<Scalar>(value, type); }
-  NodePtr MakeExpand(const Value& input0, const std::vector<int64_t>& size, const bool& is_scalar_expand) const override { return MakeNode<Expand>(input0, size, is_scalar_expand); }
-  NodePtr MakeView(const Value& input0, const std::vector<int64_t>& output_size) const override { return MakeNode<View>(input0, output_size); }
-  NodePtr MakeCast(const Value& input0, const at::ScalarType& dtype, const c10::optional<at::ScalarType>& stype = c10::nullopt) const override { return MakeNode<Cast>(input0, dtype, stype); }
-  NodePtr MakeTensorList(const OpList& inputs) const override { return MakeNode<TensorList>(inputs); }
+  NodePtr MakeScalar(const at::Scalar& value, const at::ScalarType& type) const override { return ReuseOrMakeNode<Scalar>(value, type); }
```

Comment:
hmm, no you actually do change it here. so why do you assert in the CanBeReused for scalar?